src/test/java/com/teragrep/cfe_39/HdfsTest.java

```diff
@@ -0,0 +1,260 @@
+                try {
+                    Thread.sleep(1000);
+                    hdfsReadCheck("testConsumerTopic", Integer.parseInt(lastRecord.getPartition().toString()), lastRecord.getOffset());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+
+            }
+        }
+        closeMiniCluster();
+    }
+
+    public void hdfsReadCheck(String testConsumerTopic, int partition, long offset) throws IOException {
+
+        // Check that the files were properly written to HDFS with a read test.
+        String hdfsuri = config.getHdfsuri();
+
+        String path = config.getHdfsPath()+"/"+testConsumerTopic;
+        String fileName = partition+"."+offset;
+        // ====== Init HDFS File System Object
+        Configuration conf = new Configuration();
+        // Set FileSystem URI
+        conf.set("fs.defaultFS", hdfsuri);
+        // Because of Maven
+        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        conf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+        // Set HADOOP user
+        System.setProperty("HADOOP_USER_NAME", "hdfs");
+        System.setProperty("hadoop.home.dir", "/");
+        //Get the filesystem - HDFS
+        FileSystem fs = FileSystem.get(URI.create(hdfsuri), conf);
+
+        //==== Create directory if not exists
+        Path workingDir=fs.getWorkingDirectory();
+        Path newDirectoryPath= new Path(path);
+        if(!fs.exists(newDirectoryPath)) {
+            // Create new Directory
+            fs.mkdirs(newDirectoryPath);
+            // logger.info("Path "+path+" created.");
+        }
+
+        // This is the HDFS write path for the files:
+        // Path hdfswritepath = new Path(newDirectoryPath + "/" + fileName); where newDirectoryPath is config.getHdfsPath() + "/" + lastObject.topic; and filename is lastObject.partition+"."+lastObject.offset;
+
+        //==== Read files
+        // logger.info("Read file into hdfs");
+        //Create a path
+        Path hdfsreadpath = new Path(newDirectoryPath + "/" + fileName); // The path should be the same that was used in writing the file to HDFS.
+        //Init input stream
+        FSDataInputStream inputStream = fs.open(hdfsreadpath);
+        //The data is in AVRO-format, so it can't be read as a string.
+        DataFileStream<SyslogRecord> reader = new DataFileStream<>(inputStream, new SpecificDatumReader<>(SyslogRecord.class));
+        SyslogRecord record = null;
+        int looper;
+        if (offset == 8) {
+            looper = 0;
+        } else if (offset == 13) {
+            looper = 9;
+        }else {
+            looper = 0;
+            Assertions.fail("The offset of the last record is not 8 or 13, which means a failed test.");
+        }
```

Comment:
What is the meaning of this code?

Reply:
Refactored in commit 62354239c6520133cea583da8adda051da3117e6